core/src/banking_stage.rs

```diff
@@ -1397,21 +1396,29 @@ impl BankingStage {
                 gossip_vote_sender,
             );
 
+        let mut unlock_time = Measure::start("unlock_time");
+        // Once the accounts are new transactions can enter the pipeline to process them
+        drop(batch);
+        unlock_time.stop();
+
         let ExecuteAndCommitTransactionsOutput {
             ref mut retryable_transaction_indexes,
             ref execute_and_commit_timings,
             ..
         } = execute_and_commit_transactions_output;
 
+        Self::commit_or_cancel_transaction_cost(
```

Comment:
This isn't a great way to determine what was executed or not. There are more cases where transactions aren't executed inside `execute_and_commit_transactions_locked` so we should actually update that method to return which indexes were executed or not.

Reply:
> This isn't a great way to determine what was executed or not. There are more cases where transactions aren't executed inside `execute_and_commit_transactions_locked` so we should actually update that method to return which indexes were executed or not.

Dang, you are referring to those failed but not-retryable , such as too-old?

Reply:
@taozhu-chicago Are you on this or shall I look?

Reply:
Yeah exactly, my understanding is that we don't want any non committed transactions to impact the cost

Reply:
On 2nd thought, can probably do it in separate PR? The main issues here is those retryable transactions being repeatedly applied to cost_trackers, consume its limits. The not-retryable will still take away space but only once. wdyt? @jstarry @ckamm